x-pack/plugins/cloud_defend/server/lib/telemetry/collectors/installation_stats_collector.ts

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import yaml from 'js-yaml';
+import type { CoreStart, Logger, SavedObjectsClientContract } from '@kbn/core/server';
+import type { ElasticsearchClient } from '@kbn/core-elasticsearch-server';
+import {
+  AgentPolicy,
+  PackagePolicy,
+  PACKAGE_POLICY_SAVED_OBJECT_TYPE,
+  SO_SEARCH_LIMIT,
+} from '@kbn/fleet-plugin/common';
+import { agentPolicyService } from '@kbn/fleet-plugin/server/services';
+import type { CloudDefendInstallationStats } from './types';
+import type { CloudDefendPluginStart, CloudDefendPluginStartDeps } from '../../../types';
+import { INTEGRATION_PACKAGE_NAME, INPUT_CONTROL } from '../../../../common/constants';
+import { getInputFromPolicy } from '../../../../common/utils/helpers';
+
+export const getInstallationStats = async (
+  esClient: ElasticsearchClient,
+  soClient: SavedObjectsClientContract,
+  coreServices: Promise<[CoreStart, CloudDefendPluginStartDeps, CloudDefendPluginStart]>,
+  logger: Logger
+): Promise<CloudDefendInstallationStats[]> => {
+  const [, cloudDefendServerPluginStartDeps] = await coreServices;
+
+  const cloudDefendContext = {
+    logger,
+    esClient,
+    soClient,
+    agentPolicyService: cloudDefendServerPluginStartDeps.fleet.agentPolicyService,
+    packagePolicyService: cloudDefendServerPluginStartDeps.fleet.packagePolicyService,
+  };
+
+  const getInstalledPackagePolicies = async (
+    packagePolicies: PackagePolicy[],
+    agentPolicies: AgentPolicy[]
+  ) => {
+    const installationStats = packagePolicies.map(
+      (packagePolicy: PackagePolicy): CloudDefendInstallationStats => {
+        const agentCounts =
+          agentPolicies?.find((agentPolicy) => agentPolicy?.id === packagePolicy.policy_id)
+            ?.agents ?? 0;
+
+        const input = getInputFromPolicy(packagePolicy, INPUT_CONTROL);
+        const policyYaml = input?.vars?.configuration?.value;
+        let policyJson: any = {};
+
+        try {
+          policyJson = JSON.stringify(yaml.load(policyYaml));
+        } catch (err) {
+          logger.error(
+            'could not convert cloud_defend yaml to json for installation stats usage collection',
+            err
+          );
+        }
+
+        return {
+          package_policy_id: packagePolicy.id,
+          policy_yaml: policyYaml,
```

Comment:
Hi @mitodrummer, it seems that it returns all the yaml / Json as is, which has a lot of benefits. 
However, when attempting to track the critical data on a large number of customers, parsing the critical data will be more helpful. 

For example, `sessionLeaderInteractive:  bool` 

Then you will have a detailed view of all customers and whether `sessionLeaderInteractive` is enabled or disabled and you will be able to track the usage of this field with graphs / aggregated tables etc.

Reply:
Yea, my plan was to just send up the full "json" version of this yaml policy, and then use BQ json functions to aggregate on specifics like "how many customers are using sessionLeaderInteractive". Though, it's probably worth spending some time writing some of these queries against the bronze table first to see if there are any issues with this approach.

Reply:
Thinking more on this, it would probably make sense to do some pre processing on policy yaml in kibana as we still should probably support the kibana based telemetry project (https://telemetry-v2-staging.elastic.dev/) which wouldn't have the same capabilities as BQ to further aggregate dynamic json. Having at least some statistics precalculated (e.g seletor use counts, response action counts etc..). for the policy is beneficial.

Reply:
The json version of the policy yaml is now fully "typed" in the telemetry schema. This should allow for any query on policy usage across customers. e.g how many are using 'block' action. Which types of selectors responses etc...

resolving this, but plz re-open if I missed anything.